docs/class5/connexml.md

```diff
@@ -484,6 +487,174 @@ An existing call is transferred to a different destination using the `Dial` ver
             </Dial>
         </Response>
         ```
+    15. **statusCallbackEvent, statusCallback, statusCallbackMethod**
+        ``` xml
+        <?xml version="1.0" encoding="UTF-8"?>
+        <Response>
+            <Dial>
+                <Client
+                statusCallbackEvent="initiated ringing answered completed"
+                statusCallback="http://fr1js1.connexcs.net:3002"
+                statusCallbackMethod="POST">
+                7900
+                </Client>
+            </Dial>
+        </Response>
+        ```
+        Send updates about the call's lifecycle (initiated, ringing, answered, and completed) to the callback URL http://fr1js1.connexcs.net:3002 via HTTP POST requests.
+
+#### Dynamic Dial
+
+Dynamically dial a phone number based on a `substring` of a variable named `Extension`.
+
+!!! Example "Example 1"
+    ```xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <Response>
+        <Dial>
+            <Number>>{% raw %}{{ substring Extension 5 }}{% endraw %}</Number>
+        </Dial>
+    </Response>
+    ```
+    **Purpose**: Initiates a call to a number extracted from the Extension variable.
+
+    **Actions**: Extracts a substring from the Extension variable starting at the 5th character and uses it as the number to dial.
+
+!!! Example "Example 2"
+    Assuming the `Extension variable` is set to `"1234567890"`:
+    The `substring` starting at the `5th character` is `"67890"`.
+    The system will dial the number `"67890"`.
+
+!!! Example "Example 3"
+    Assuming the `Extension variable` is set to `"afwd_[a-z0-9]+"`:
+    It means if you dial `newsetup_160` and
+    ```xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <Response>
+        <Dial>
+            <Number>{% raw %}{{ substring Extension 9 }}{% endraw %}</Number>
+        </Dial>
+    </Response>
+    ```
+    The system will dial the number `"160"`
+
+### Press DTMF Variable
+
+It helps to define on which leg of the call the DTMF will work. For example, `dtmf_leg ='a'` or `dtmf_leg ='b'`.
+
+!!! Example
+    When the digit `3` is pressed on the call leg `b` within the specified context `A`, the call will be transferred to `test4`.
+    ``` xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <Response>
+        <Press dtmf_leg ='b'digit="3" context="A">
+            <Transfer>test4</Transfer>
+        </Press>
+    </Response>
+    ```
+
+|**Noun**|**Description**|
+|--------|---------------|
+|`Transfer`|Transfers the call to the given extension given|
+
+### Before
+
+It initializes variables before executing other commands.
+
+!!! Example
+    ``` xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <Response>
+        <Before><Set name="newHome" value="Adam"/><Set name="headerName" value="Joe"/></Before>
+    </Response>
+    ```
+    Sets the variable `newHome` to "**Adam**".<br>Sets the variable `headerName` to "**Joe**"</br>.
+
+### Stream
```

Comment:
add more details to stream like parameters
<Response>
	<Start>
		<Stream url="ws://fr1js1.connexcs.net:3001">
			<Parameter name="Hello" value="how are you, " />
		</Stream>
	</Start>

Reply:
|**Attribute**|**Description**|**Options**|**Default Method**|
|-------------|---------------|-----------|-------|
|`url`|The WebSocket destination address for stream delivery||none|
|`name`(Optional)|Defines a unique custom name for the stream instance||none|
|`track`(Optional)|Indicates which track to stream|`inbound_track`, `outbound_track`, `both_tracks`|`inbound_track`|
|`statusCallback`|For every event listed in the `statusCallbackEvent` attribute, you may define a URL to send webhook requests to using the `statusCallback` attribute. A valid hostname is required for non-relative URLs (underscores are not allowed)|any `URL`|none|
|`statusCallbackMethod`|Lets you define the `HTTP` method ConnexCS should use when making requests to the URL specified in the `statusCallback` attribute.|`GET`, `POST`|`POST`|
|`bidirectionalMode`|Bidirectional streaming mode|`mp3`, `rtp`|`mp3`|
|`bidirectionalCodec`|Bidirectional streaming codec, used only with `bidirectionalMode=rtp`|`PCMU`, `PCMA`, `G722`|`PCMU`|